source/fundamentals/connection.txt

```diff
@@ -0,0 +1,636 @@
+
+.. _compression-dependencies:
+
+Compression Algorithm Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The JDK supports `Zlib <https://zlib.net/>`__ compression natively, but
+`Snappy <https://google.github.io/snappy/>`__ and
+`Zstandard <https://github.com/facebook/zstd/>`__ depend upon open source
+implementations. See 
+`snappy-java <https://github.com/xerial/snappy-java>`__ and
+`zstd-java <https://github.com/luben/zstd-jni>`__ for details.
+
+.. _connect-replica-set:
+
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration of instances provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname and port numbers
+of one or more members of the replica set separated by commas.
+
+.. code-block:: none
+
+   mongodb://host1:27017,host2:27017,host3:27017/?replicaSet=myRs
+
+.. note::
+
+   Although it is possible to connect to a replica set deployment by
+   providing only a single host, you should provide the driver with the
+   full list to ensure that it is able to connect even if one host fails.
+
+.. _connection-options:
+
+Connection Options
+------------------
+
+This section explains MongoDB connection and authentication options
+supported by the Java driver. You can pass the connection options as
+parameters of the connection URI to specify the behavior of the client.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option Name
+     - Type
+     - Description
+
+   * - **maxPoolSize**
+     - integer
+     - Specifies the maximum size of the instance connection pool.
+
+   * - **waitQueueTimeoutMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, that a
+       thread may wait for a connection to become available.
+
+   * - **serverSelectionTimeoutMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the driver
+       will wait for server selection to succeed before throwing an
+       exception.
+
+   * - **localThresholdMS**
+     - integer
+     - When communicating with multiple instances of MongoDB in a replica
+       set, the Java driver will only send requests to a server whose
+       response time is less than or equal to the server with the fastest
+       response time plus the local threshold, in milliseconds.
+
+   * - **heartbeatFrequencyMS**
+     - integer
+     - Specifies the frequency, in milliseconds, that the driver will
+       wait between attempts to determine the current state of each
+       server in the cluster.
+
+   * - **replicaSet**
+     - string
+     - Specifies that the :ref:`connection string <connection-uri>`
+       provided includes multiple hosts. When specified, the Java driver
+       attempts to find all members of that set.
+
+   * - **ssl**
+     - boolean
+     - Specifies that all communication with MongoDB instances should
+       use TLS.
```

Comment:
I've only included it since it's mentioned by name in the **tls** option, but I've added an extra note that this is superseded by **tls** to encourage users to use tls instead.